rpc/types_transaction_receipt.go

```diff
@@ -358,48 +189,66 @@ type TxnStatusResp struct {
 }
 
 type TransactionReceiptWithBlockInfo struct {
-	UnknownTransactionReceipt
+	TransactionReceipt
 	BlockHash   *felt.Felt `json:"block_hash,omitempty"`
 	BlockNumber uint       `json:"block_number,omitempty"`
 }
 
-func (t *TransactionReceiptWithBlockInfo) UnmarshalJSON(data []byte) error {
-	var uTxnRec UnknownTransactionReceipt
-	err := uTxnRec.UnmarshalJSON(data)
-	if err != nil {
-		return err
-	}
-	t.UnknownTransactionReceipt = uTxnRec
-
+func (t *TransactionReceiptWithBlockInfo) MarshalJSON() ([]byte, error) {
 	aux := &struct {
+		TransactionReceipt
 		BlockHash   string `json:"block_hash,omitempty"`
 		BlockNumber uint   `json:"block_number,omitempty"`
-	}{}
+	}{
+		TransactionReceipt: t.TransactionReceipt,
+		BlockHash:          t.BlockHash.String(),
+		BlockNumber:        t.BlockNumber,
+	}
+
+	return json.Marshal(aux)
+}
+
+func (tr *TransactionReceiptWithBlockInfo) UnmarshalJSON(data []byte) error {
+	var aux struct {
+		TransactionHash    *felt.Felt         `json:"transaction_hash"`
+		ActualFee          FeePayment         `json:"actual_fee"`
+		ExecutionStatus    TxnExecutionStatus `json:"execution_status"`
+		FinalityStatus     TxnFinalityStatus  `json:"finality_status"`
+		Type               TransactionType    `json:"type,omitempty"`
+		MessagesSent       []MsgToL1          `json:"messages_sent"`
+		RevertReason       string             `json:"revert_reason,omitempty"`
+		Events             []Event            `json:"events"`
+		ExecutionResources ExecutionResources `json:"execution_resources"`
+		ContractAddress    *felt.Felt         `json:"contract_address,omitempty"`
+		MessageHash        NumAsHex           `json:"message_hash,omitempty"`
+		BlockHash          string             `json:"block_hash,omitempty"`
+		BlockNumber        uint               `json:"block_number,omitempty"`
+	}
 
 	if err := json.Unmarshal(data, &aux); err != nil {
 		return err
 	}
+
+	tr.TransactionReceipt = TransactionReceipt{
```

Comment:
And then, we can also change it to something like this to avoid unnecessary code:

`tr.TransactionReceipt = aux.TransactionReceipt`

Reply:
Done